lib/puppet/type/cisco_bfd_global.rb

```diff
@@ -0,0 +1,294 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_bfd_global) do
+  @doc = "
+    Manages the Cisco Bfd Global configuration resource.
+    cisco_bfd_global {'default':
+      ..attributes..
+    }
+    'default' is only acceptable name for this global config object.
+    Example:
+    cisco_bfd_global { 'default':
+      ensure                => 'present',
+      echo_interface        => 'loopback10',
+      echo_rx_interval      => 300,
+      fabricpath_interval   => ['750', '350', '45'],
+      fabricpath_slow_timer => 15000,
+      fabricpath_vlan       => 100,
+      interval              => ['100', '100', '25'],
+      ipv4_echo_rx_interval => 100,
+      ipv4_interval         => ['200', '200', '50'],
+      ipv4_slow_timer       => 10000,
+      ipv6_echo_rx_interval => 200,
+      ipv6_interval         => ['500', '500', '30'],
+      ipv6_slow_timer       => 25000,
+      slow_timer            => 5000,
+      startup_timer         => 25,
+    }
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  newparam(:name, namevar: :true) do
+    desc "Instance of bfd global, only allow the value 'default'"
+
+    validate do |name|
+      if name != 'default'
+        error "only 'default' is accepted as a valid bfd_global name"
+      end
+    end
+  end # param id
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:echo_interface) do
+    desc "Loopback interface used for echo frames. Valid values are
+          string, keyword 'default'."
+
+    munge do |value|
+      value = :default if value == 'default'
+      value
+    end
+  end # property echo_interface
+
+  newproperty(:echo_rx_interval) do
+    desc "Echo receive interval in msec. Valid values are
+          integer, keyword 'default'."
+
+    munge do |value|
+      value = :default if value == 'default'
+      begin
+        value = Integer(value) unless value == :default
+      rescue
+        raise 'echo_rx_interval must be a valid integer, or default.'
+      end
+      value
+    end
+  end # property echo_rx_interval
+
+  newproperty(:fabricpath_interval, array_matching: :all) do
+    format = '[fabricpath_interval, fabricpath_min_rx, fabricpath_multiplier]'
+    desc 'An array of [fabricpath_interval, fabricpath_min_rx, fabricpath_multiplier]'\
+         "Valid values match format #{format}."
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
```

Comment:
You don't need the `begin / end`. Just use a ternary.

`munge { |value| value == 'default' ? :default : value }`